health-checker/src/main/java/io/stargate/health/HealthCheckerServerFactory.java

```diff
@@ -1,57 +1,12 @@
 package io.stargate.health;
 
-import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.health.HealthCheckRegistry;
-import com.codahale.metrics.servlets.HealthCheckServlet;
 import com.fasterxml.jackson.annotation.JsonTypeName;
-import io.dropwizard.jersey.filter.AllowedMethodsFilter;
-import io.dropwizard.jetty.MutableServletContextHandler;
-import io.dropwizard.jetty.NonblockingServletHolder;
 import io.dropwizard.server.SimpleServerFactory;
-import java.util.EnumSet;
-import java.util.stream.Collectors;
-import javax.servlet.DispatcherType;
-import org.eclipse.jetty.server.Handler;
-import org.eclipse.jetty.server.Server;
 
 /**
- * Custom DropWizard server factory, in order to plug our {@link HealthCheckerAdminServlet}.
- *
- * <p>This class is made available to DropWizard via {@code
+ * This class is made available to DropWizard via {@code
  * META-INF/services/io.dropwizard.server.ServerFactory}, and then enabled via the {@code
  * server.type} option in {@code config.yaml}.
  */
 @JsonTypeName("health-checker")
-public class HealthCheckerServerFactory extends SimpleServerFactory {
-
-  @Override
-  protected Handler createAdminServlet(
-      Server server,
-      MutableServletContextHandler handler,
-      MetricRegistry metrics,
-      HealthCheckRegistry healthChecks) {
-    configureSessionsAndSecurity(handler, server);
-    handler.setServer(server);
-
-    handler
-        .getServletContext()
-        .setAttribute(HealthCheckServlet.HEALTH_CHECK_REGISTRY, healthChecks);
-    handler.addServlet(new NonblockingServletHolder(new HealthCheckerAdminServlet()), "/*");
-    final String allowedMethodsParam =
-        getAllowedMethods().stream().collect(Collectors.joining(","));
-    handler
-        .addFilter(AllowedMethodsFilter.class, "/*", EnumSet.of(DispatcherType.REQUEST))
-        .setInitParameter(AllowedMethodsFilter.ALLOWED_METHODS_PARAM, allowedMethodsParam);
-    return handler;
-  }
-
-  private void configureSessionsAndSecurity(MutableServletContextHandler handler, Server server) {
-    handler.setServer(server);
-    if (handler.isSecurityEnabled()) {
-      handler.getSecurityHandler().setServer(server);
-    }
-    if (handler.isSessionsEnabled()) {
-      handler.getSessionHandler().setServer(server);
-    }
-  }
-}
+public class HealthCheckerServerFactory extends SimpleServerFactory {}
```

Comment:
Do we still need to have this? Could we use the dropwizard's default server factory?

Reply:
The class is annotated with `@JsonTypeName("health-checker")`, does this play any role? I was not certain, thus kept it..

Reply:
It is referenced in `health-checker/src/main/resources/META-INF/services/io.dropwizard.server.ServerFactory`. If we control DW factories from our code now, it should not be necessary anymore, I guess :thinking:

Reply:
I am leaving it for now, as I see other modules do the same.

Reply:
sgtm :+1: